### PR TITLE
Test review E-I, #259

### PR DESCRIPTION
**Pull request by paulirwin**
- [X] You've read the [Contributor Guide](https://github.com/apache/lucenenet/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://www.apache.org/foundation/policies/conduct.html).
- [X] You've included unit or integration tests for your change, where applicable.
- [X] You've included inline docs for your change, where applicable.
- [X] There's an open issue for the PR that you are making. If you'd like to propose a change, please [open an issue](https://github.com/apache/lucenenet/issues/new/choose) to discuss the change or find an existing issue.

Test review for core E-I tests

Partial #259 

## Description

This updates the tests in the E-I test assembly to match upstream Lucene for style, formatting, etc. Additionally, some minor bugs (including documentation bugs) have been fixed.

### CODE REVIEW

**Comment by paulirwin** [comment]
@NightOwl888 Updated title to indicate the breaking changes of changing some static readonly fields to be const. This will affect any NuGet packages or custom binaries that use these fields, although it's generally not breaking from a source standpoint. We should probably do a sweep before beta 18 for other `const`-able fields.

**Comment by NightOwl888** [comment]
> @NightOwl888 Updated title to indicate the breaking changes of changing some static readonly fields to be const. This will affect any NuGet packages or custom binaries that use these fields, although it's generally not breaking from a source standpoint. We should probably do a sweep before beta 18 for other `const`-able fields.

Looks like this only affects test classes except for the `Assert` class that is part of the test framework, but there are no breaking API or behavioral changes in it. Or am I missing something?

**Comment by paulirwin** [comment]
IndexWriter and ByteBlockPool have some public const changes. I can revert those if we'd rather do them as part of a separate sweep.

**Comment by NightOwl888** [comment]
> IndexWriter and ByteBlockPool have some public const changes. I can revert those if we'd rather do them as part of a separate sweep.

I see 'em. I have 7 files left to review and those 2 are at the bottom.

**Comment by NightOwl888** [comment]
If these are made const, that means they will be compiled as metadata in the consuming assembly. So any changes to the values will not be seen unless the consuming app is recompiled. 

This may or may not be a problem, though. Technically, we should never update the values on these unless porting a new version of Lucene. It would be helpful to know if Lucene ever changes values during minor or patch version bumps, though, because we will eventually catch up with them. The safest bet without knowing that is to leave them `static`. That said, I have been guilty of changing many of these to `const` recently.

**Review by NightOwl888 — CHANGES_REQUESTED** [review]
Nice work!

I found a few minor things to address. Please see my comments inline.

File: src/Lucene.Net.Tests/Index/TestAddIndexes.cs [hunk 1]
```diff
@@ -665,7 +741,7 @@ public RunAddIndexesThreads(TestAddIndexes outerInstance, int numCopy)
                 }
             }
 
-            internal virtual void LaunchThreads(int numIter)
+            internal void LaunchThreads(int numIter)
```
[the remark this answers is not in the document]
Comment (paulirwin) [reply src/Lucene.Net.Tests/Index/TestAddIndexes.cs:744]: This is in a private class, but will do.

File: src/Lucene.Net.Tests/Index/TestBinaryDocValuesUpdates.cs [hunk 2]
```diff
@@ -1142,46 +1137,17 @@ public virtual void TestUpdateOldSegments()
...
-            var writer = new IndexWriter(dir, conf);
-            writer.AddDocument(doc);
+            conf = NewIndexWriterConfig(TEST_VERSION_CURRENT, new MockAnalyzer(Random));
+            writer = new IndexWriter(dir, conf);
             writer.UpdateBinaryDocValue(new Term("id", "doc"), "f", ToBytes(4L));
-
-            try
-            {
+            OldFormatImpersonationIsActive = false;
+            try {
```
[the remark this answers is not in the document]
Comment (paulirwin) [reply src/Lucene.Net.Tests/Index/TestBinaryDocValuesUpdates.cs:1145]: That was unintentional, will fix.

File: src/Lucene.Net.Tests/Index/TestDirectoryReaderReopen.cs [hunk 3]
```diff
@@ -393,7 +386,7 @@ public override void Run()
                     if (index % 2 == 0)
                     {
                         // refresh reader synchronized
-                        ReaderCouple c = (outerInstance.RefreshReader(r, test, index, true));
+                        ReaderCouple c = outerInstance.RefreshReader(r, test, index, true);
```
[the remark this answers is not in the document]
Comment (paulirwin) [reply src/Lucene.Net.Tests/Index/TestDirectoryReaderReopen.cs:389]: Agreed, I'm all in favor of making methods static where possible. Slightly better performance of not having to load the `this` argument (as well as all the other `outerInstance` use). I went ahead and did that for two other methods in this class as well.

**Comment by paulirwin** [comment]
@NightOwl888 All feedback items should be addressed now. Thanks!

File: src/Lucene.Net.Tests/Index/TestBinaryDocValuesUpdates.cs [hunk 4]
```diff
@@ -1142,7 +1142,8 @@ public virtual void TestUpdateOldSegments()
             writer = new IndexWriter(dir, conf);
             writer.UpdateBinaryDocValue(new Term("id", "doc"), "f", ToBytes(4L));
             OldFormatImpersonationIsActive = false;
-            try {
+            try
+            {
                 writer.Dispose();
                 Assert.Fail("should not have succeeded to update a segment written with an old Codec");
             } catch (Exception e) when (e.IsUnsupportedOperationException()) {
```
Comment (NightOwl888) [inline src/Lucene.Net.Tests/Index/TestBinaryDocValuesUpdates.cs:1149]: I am guessing you intended to fix this line `} catch (Exception e) when (e.IsUnsupportedOperationException()) {` and `} finally {` also?

**Review by NightOwl888 — APPROVED** [review]

**Pull request merged**